Rename echconfig to ech in SVCB/HTTPS records

adam-stoler opened the pull request:
In the latest SVCB RFC draft (https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https-06), the parameter "echconfig" has been renamed to "ech".  This Pull Request implements that change for dnsjava, and renames the classes and variables appropriately.

ibauersachs reviewed (changes requested):
The changes break semantic versioning if I were to apply them without going to v4, and a rename is not significant enough for that.
We could either duplicate the `ParameterEchConfig` class and deprecate the existing, or keep the class name and only add `public static final int ECH = 5;` (but also keeping `ECHCONFIG` as deprecated).

adam-stoler commented:
Good point.  I was thinking that these were internal classes and values, but they are public so must remain consistent in the API.  I went the approach of duplicating and deprecating the old ones.  Let me know if you think that anything else is needed.

ibauersachs commented:
Actually yes, could you please add Javadoc with `@deprecated` tags mentioning what to use instead?
I also added `@since` tags to all new stuff added in or after 3.0, so if you wouldn't mind...

adam-stoler commented:
Sure, I added the javadoc @deprecated tags with links to the correct ones to use and added @since tags to the SVCB and HTTPS record classes.  I also made SVCBRecordBase public so it and its nested classes would show up in javadoc.

ibauersachs commented:
Thanks!
I was looking at this from my mobile so far and missed that e.g. `1 h3pool. alpn=h2,h3 echconfig=1234` is now not parseable anymore AFAICT. My gut feeling however says that we should be able to since this is what was produced until this patch. But you may convince me otherwise.

adam-stoler commented:
I was torn on this issue.  I generally think it is a good idea to maintain backwards compatibility, but since this is still not even an official RFC, I was not sure it was worth the effort to support the obsolete name.  Though now that I see how easy this is to support with the alias functionality in the Mnemonic class, it seems worthwhile to support this still just in case.

Note that with the implemented solution, it will always write out the RDATA with the "ech" name even if it was parsed from a string that used the obsolete "echconfig" name.  I think that is desirable (and simpler to implement), but let me know if you disagree.

ibauersachs commented:
> I was torn on this issue. I generally think it is a good idea to maintain backwards compatibility, but since this is still not even an official RFC, I was not sure it was worth the effort to support the obsolete name. Though now that I see how easy this is to support with the alias functionality in the Mnemonic class, it seems worthwhile to support this still just in case.

It's probably something that can be removed in v4 if the RFC is final.

> Note that with the implemented solution, it will always write out the RDATA with the "ech" name even if it was parsed from a string that used the obsolete "echconfig" name. I think that is desirable (and simpler to implement), but let me know if you disagree.

IMO that's the preferred way.

adam-stoler commented:
Sounds great, thanks!
